tests/periph_timer_short_relative_set/main.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2019 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Peripheral timer test application
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "board.h"
+#include "thread.h"
+#include "thread_flags.h"
+
+#include "periph/timer.h"
+
+#ifndef TEST_TIMER_DEV
+# include "xtimer.h"
+# define TEST_TIMER_DEV      XTIMER_DEV
+# define TEST_TIMER_FREQ     XTIMER_HZ
+#else
+# ifndef TEST_TIMER_FREQ
+#  define TEST_TIMER_FREQ     (1000000LU)
+# endif
+#endif
+
+#ifndef TEST_MAX_DIFF
+#define TEST_MAX_DIFF   (1000LU)
+#endif
+
+static void cb(void *arg, int chan)
+{
+    (void)chan;
+    thread_flags_set(arg, 1);
+}
+
+int main(void)
+{
+    puts("\nTest for peripheral TIMER short timer_set()\n");
+
+    printf("This test tries timer_set() with decreasing intervals down to 0.\n"
+           "You should see lines like 'interval <n> ok', followed by a success"
+           " message.\n"
+           "On failure, this test prints an error message.\n\n");
+
+    printf("testing periph_timer %u, freq %lu\n", TEST_TIMER_DEV, TEST_TIMER_FREQ);
+    timer_init(TEST_TIMER_DEV, TEST_TIMER_FREQ, cb, (thread_t *)sched_active_thread);
+
+    uint32_t interval = 100;
+    while (interval--) {
+        uint32_t before = timer_read(TEST_TIMER_DEV);
+        timer_set(TEST_TIMER_DEV, 0, interval);
+        while(!thread_flags_clear(1)) {
+            uint32_t diff = timer_read(TEST_TIMER_DEV) - before;
```

Comment:
this assumes timer bit width is equal to platform bit width, e.g. 32 bit on 32 bit platform. However this will be wrong when using 16bit timer on a 32 bit platform, for example. See #9283 and #10493.

Reply:
Thanks for jumping in! Yes, this calculation doesn't give the correct value for `diff`. The reason it does work in this test is because testing all values can be done in the first period before an overflow happens - but with 8 bit timers even that won't work.
Seeing the need for proper diff operation again ...I think I should have a look at #10493 soonish :)